key-manager/internal/models/kserve.go

```diff
@@ -78,3 +79,36 @@ func (m *Manager) ListAvailableModels() ([]ModelInfo, error) {
 
 	return modelList, nil
 }
+
+// ListAvailableLLMs lists all LLMInferenceServices across all namespaces.
+func (m *Manager) ListAvailableLLMs() ([]openai.Model, error) {
+	// Define InferenceService GVR
+	llmGVR := schema.GroupVersionResource{
+		Group:    "serving.kserve.io",
+		Version:  "v1alpha1",
+		Resource: "llminferenceservices",
+	}
+
+	log.Printf("DEBUG: Attempting to list LLMInferenceServices with GVR: %+v", llmGVR)
```

Comment:
Long-term, we need a better logging strategy so that we can have different levels, for example.